todo_app/app.py

```diff
@@ -25,9 +48,20 @@ def index():
             items.sort(key=lambda x: x.due_date)
         elif request.values.get('sort') == '2':
             items.sort(key=lambda x: x.due_date, reverse=True)
-        return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+
+        if hasattr('current_user', 'role'):
+            if current_user.role == 'writer':
+                return render_template("index.html",View_Model=get_view_model, todays_date=todays_date)
+            elif current_user.role == 'reader':
+                return render_template('index_read.html', View_Model=get_view_model, todays_date=todays_date)
+            else:
```

Comment:
Is defaulting to the write-template sensible behaviour? If a new role is introduced and someone forgets to update this code, then we're going to risk exposing functionality that they shouldn't have